packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
+  labelId?: string;
+  dataTestId?: string;
+  description?: React.ReactChild;
+  value: number;
+  min?: number;
+  step?: number;
+  max?: number;
+  onCountChange?(count: number): void;
+  error?: React.ReactNode | React.ReactNode[];
 }
 
-const val = 5;
-
-const handleChange = jest.fn((num) => num);
-
-const requiredAttributes = {
-  value: val,
-  onCountChange: handleChange,
-};
-
-const counterMock = ({
+const CounterMock: React.FC<MockProps> = ({
   ref,
   label = 'Label',
   labelId = '',
-  id = '',
   error = '',
+  id = '',
   description = '',
-  value,
-  onCountChange,
-  min = 0,
-  max = 10,
+  value = 5,
   dataTestId = '',
-}: MockCounterProps) => (
-  <Counter
-    ref={ref}
-    label={label}
-    labelId={labelId}
-    error={error}
-    id={id}
-    description={description}
-    value={value}
-    data-testid={dataTestId}
-    min={min}
-    max={max}
-    onCountChange={onCountChange}
-  />
-);
+  min = 0,
+  max = 100,
+  onCountChange,
+}) => {
+  const [counterValue, setCounterValue] = useState(value);
+  const handleChange = (value: number) => setCounterValue(value);
+
+  return (
+    <Counter
+      ref={ref}
+      label={label}
+      labelId={labelId}
+      error={error}
+      id={id}
+      description={description}
+      value={counterValue}
+      data-testid={dataTestId}
+      min={min}
+      max={max}
+      onCountChange={onCountChange ?? handleChange}
+    />
+  );
+};
 
 test('forwards ref', () => {
   const ref = createRef<HTMLInputElement>();
-  const { container } = render(counterMock({ ref, ...requiredAttributes }));
-  const counter = container.querySelector('input');
 
-  expect(ref.current).toBe(counter);
+  render(<Counter value={5} onCountChange={jest.fn()} ref={ref} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
+
+  expect(ref.current).toBe(input);
 });
 
 test('renders a counter as an input tag', () => {
-  const { container } = render(counterMock(requiredAttributes));
+  render(<CounterMock value={5} />);
 
-  expect(container.querySelector('input')).toBeInTheDocument();
+  expect(screen.getByLabelText(/Label/i)).toBeInTheDocument();
 });
 
 test('renders a counter with matched label', () => {
-  const { queryByLabelText } = render(counterMock({ label: 'Test Label', ...requiredAttributes }));
+  render(<CounterMock value={5} label="Test Label" />);
 
   // This one checks for matching id and htmlFor
-  expect(queryByLabelText('Test Label')).toBeInTheDocument();
+  expect(screen.getByLabelText(/test label/i)).toBeInTheDocument();
 });
 
 test('create unique ids if not provided', () => {
-  const { queryByTestId } = render(
+  render(
     <>
-      {counterMock({ dataTestId: 'item1', label: 'Test Label', ...requiredAttributes })}
-      {counterMock({ dataTestId: 'item2', label: 'Test Label', ...requiredAttributes })}
+      {<CounterMock value={5} dataTestId="item1" label="Test Label" />}
+      {<CounterMock value={5} dataTestId="item2" label="Test Label" />}
```

Comment:
`{}` are only required when calling a method, in this case we are just rendering components:


```suggestion
      <CounterMock value={5} dataTestId="item1" label="Test Label" />
      <CounterMock value={5} dataTestId="item2" label="Test Label" />
```